### ISSUE TITLE
Heroku deploy not working

### ISSUE THREAD

**Issue by michaelmior** [NodeBB/NodeBB#5792]
I followed [the instructions](https://docs.nodebb.org/installing/cloud/heroku/) for deploying on Heroku, but the app fails to load. Using v1.5.1@e64104786 with redis.

First, I was encountering an error that removing the PID file was failing because it didn't exist. Odd, but I just decided to put an `existsSync` check to see what would happen. Still fails and when I set `NODE_ENV` to `development`, I get the following error:

```
Error: Failed to lookup view "500" in views directory "/app/build/public/templates"
    at Function.render (/app/node_modules/express/lib/application.js:580:17)
    at ServerResponse.render (/app/node_modules/express/lib/response.js:971:7)
    at content (/app/src/middleware/render.js:70:15)
    at /app/node_modules/async/dist/async.js:3830:24
    at replenish (/app/node_modules/async/dist/async.js:946:17)
    at /app/node_modules/async/dist/async.js:950:9
    at eachOfLimit (/app/node_modules/async/dist/async.js:975:24)
    at /app/node_modules/async/dist/async.js:980:16
    at eachOf (/app/node_modules/async/dist/async.js:1051:5)
    at _parallel (/app/node_modules/async/dist/async.js:3829:5)
    at Object.parallelLimit [as parallel] (/app/node_modules/async/dist/async.js:3912:5)
    at async.waterfall.str (/app/src/middleware/render.js:65:12)
    at nextTask (/app/node_modules/async/dist/async.js:5273:14)
    at next (/app/node_modules/async/dist/async.js:5280:9)
    at /app/node_modules/async/dist/async.js:906:16
    at fireFilterHook (/app/src/plugins/hooks.js:110:11)
```

**Comment by pitaj** [comment]
What does `./nodebb build` give you?

**Comment by michaelmior** [comment]
```
[deprecated] `String.prototype.rtrim` is deprecated as of NodeBB v1.5; use `utils.rtrim` instead.
30/6 15:27:13 [31256] - info: [build] Building in parallel mode
30/6 15:27:13 [31256] - info: [build]         plugin static dirs  build started
30/6 15:27:13 [31256] - info: [build]          requirejs modules  build started
30/6 15:27:13 [31256] - info: [build]           client js bundle  build started
30/6 15:27:13 [31256] - info: [build]            admin js bundle  build started
30/6 15:27:13 [31256] - info: [build]         client side styles  build started
30/6 15:27:13 [31256] - info: [build] admin control panel styles  build started
30/6 15:27:13 [31256] - info: [build]                  templates  build started
30/6 15:27:13 [31256] - info: [build]                  languages  build started
30/6 15:27:13 [31256] - info: [build]                     sounds  build started
30/6 15:27:14 [31256] - info: [build]                     sounds  build completed in 0.657sec
30/6 15:27:14 [31256] - info: [build]         plugin static dirs  build completed in 1.307sec
30/6 15:27:16 [31256] - info: [build]                  templates  build completed in 3.454sec
30/6 15:27:17 [31256] - info: [build]            admin js bundle  build completed in 4.537sec
30/6 15:27:17 [31256] - info: [build]                  languages  build completed in 4.596sec
30/6 15:27:19 [31256] - info: [build] admin control panel styles  build completed in 5.686sec
30/6 15:27:19 [31256] - info: [build]         client side styles  build completed in 5.824sec
30/6 15:27:19 [31256] - info: [build]           client js bundle  build completed in 6.065sec
30/6 15:27:21 [31256] - info: [build]          requirejs modules  build completed in 7.674sec
30/6 15:27:21 [31256] - info: [build] Asset compilation successful. Completed in 7.68sec.
```

**Comment by michaelmior** [comment]
Ah, there's the culprit. If I run the build locally and then commit the generated files, everything works fine. The issue is that the build never runs on Heroku. Perhaps the instructions are outdated?

**Comment by michaelmior** [comment]
As a less messy way of fixing things, adding `./nodebb build` as a script named `postinstall` to `package.json` also solves the problem.

**Comment by milesflo** [comment]
@michaelmior What did you put into the fields at startup? I'm trying to deploy it as well and I'm having trouble figuring out what the docs are telling me to do.

**Comment by michaelmior** [comment]
@pitaj Any reason this script couldn't be added to the repository? It's pretty minor and would save a step for Heroku deploys.

@milesflo What fields are you having trouble filling out?

**Comment by milesflo** [comment]
Oh, thank you for the quick reply. 

I'm on step 9.. I have a Heroku app, but I am not sure what information to put into the fields as I go through setup (pre-deployment). I've tried matching up the text in grey to what I assumed they were associated with, as well as the config that worked for local deployment.

* Another thing that happens is I'll deploy it and access the home page from the project's URL... I'll see fields similar to that of normal CL setup, but once I submit the form I get a success message that hangs, with a "Go to Page" button that does not do anything. I suspect it's an issue accessing the database on the backend, but addressing the problem above will probably fix this.

Thanks. Apologies for the length.

**Comment by michaelmior** [comment]
Use `heroku config:get REDISTOGO_URL` to get the connection info with the Redis server.

**Comment by pitaj** [comment]
@michaelmior there is a reason

We have to install modules in order for `./nodebb setup` to work. If we add the postinstall script, then it will error after every run because `./nodebb build` will fail if `./nodebb setup` isn't run first.

**Comment by julianlam** [comment]
@pitaj @michaelmoir A post install script is a good idea if it works, however we won't include it into the repo. It'd be something we include in the Heroku docs to create and check in (much like the Procfile)

**Comment by michaelmior** [comment]
Unfortunate that this won't be included in the repo, but understandable. It would be great to update the instructions though.

**Comment by michaelmior** [comment]
@pitaj It would introduce some extra complication, but you could of course check if `./nodebb setup` has been run and if not, skip the build. I know this starts to do a lot to accomodate Heroku though. Personally though I love the one click deploy experience the Heroku button gives when a repo is configured to be able to run Heroku without modification.

**Comment by milesflo** [comment]
@michaelmior heroku config:get REDISTOGO_URL yielded an empty line

**Comment by pitaj** [comment]
I'm reopening this just as a reminder for the docs

**Comment by michaelmior** [comment]
@milesflo Did you do step 8 first?

**Comment by milesflo** [comment]
@michaelmior Ahhh. I deleted the heroku project, restarted, and forgot that step. It's now giving me a URL. What did you end up doing with this URL?

**Comment by michaelmior** [comment]
@milesflo That URL contains the connection information for the Redis server.

**Comment by julianlam** [comment]
Fixed via a4054ce03f4466a139b23090abc1bbdfba49047f
